File: common/src/main/java/net/particify/arsnova/common/uuid/StringToUuidJsonConverter.java
package net.particify.arsnova.common.uuid;

import java.util.UUID;
import tools.jackson.databind.JavaType;
import tools.jackson.databind.type.TypeFactory;
import tools.jackson.databind.util.StdConverter;

public class StringToUuidJsonConverter extends StdConverter<String, UUID> {
  @Override
  public UUID convert(final String value) {
    return UuidHelper.stringToUuid(value);
  }

  @Override
  public JavaType getInputType(final TypeFactory typeFactory) {
    return typeFactory.constructType(String.class);
  }

  @Override
  public JavaType getOutputType(final TypeFactory typeFactory) {
    return typeFactory.constructType(UUID.class);
  }
}

File: authz/src/main/kotlin/net/particify/arsnova/authz/model/serialization/StringToUuidJsonConverter.kt
package net.particify.arsnova.authz.model.serialization

import com.fasterxml.jackson.databind.JavaType
import com.fasterxml.jackson.databind.type.TypeFactory
import com.fasterxml.jackson.databind.util.Converter
import java.util.UUID

class StringToUuidJsonConverter : Converter<String, UUID?> {
  override fun convert(value: String): UUID? {
    return UuidHelper.stringToUuid(value)
  }

  override fun getInputType(typeFactory: TypeFactory): JavaType {
    return typeFactory.constructType(String::class.java)
  }

  override fun getOutputType(typeFactory: TypeFactory): JavaType {
    return typeFactory.constructType(UUID::class.java)
  }
}
